javascript/README.md

```diff
@@ -334,40 +337,48 @@ function getFullName({ firstName, lastName }) {
 
 - ESLint: [prefer-destructuring]
 - JSPerf: [Object Destructuring vs Not]
+- MDN Web Docs: [Object Destructuring]
 
 ### 4.2 Array Destructuring
 
-Avoid [destructuring arrays][Array Destructuring].
+Use [array destructuring][Array Destructuring] when assigning array values to a variable.
```

Comment:
I'm not sure if we should have a rule about using array destructuring. In the past, I have worked on projects with the ESLint `prefer-destructuring` rule enabled, but sometimes I would find that it wanted me to use array destructuring even when I thought that using an index was more readable.

For example, I find `const caleb = users[0]` more readable than `const [caleb] = users`. This especially comes into play when you are getting a different item of the array, where ESLint will want you to do something like `const [,,,caleb] = users` instead of `const caleb = users[3]`. I think array destructuring is more readable only when you are retrieving multiple items that are next to each other in the array, and near index 0.

Reply:
I would agree with @calebeby on this. I don't have a good grasp on perf differences between ES6 approaches and older, still valid ways of doing things, but I'm a little wary of preventing users from deciding which approach works best for a particular scenario. Even with ES5, for example, I almost alway prefer to use Array extras for iterating over an array (or an Object's keys), but there may be times when I want to use a loop (with `let` instead of `var`) for some reason. If we use warnings or, IDK - does ESLint even have something like "suggestions"? - instead of errors, I guess I may be ok with that.

Reply:
Excellent points @calebeby & @Josh68! 

Just to make sure I'm clear, the suggestion here is to remove the "**4.2 Array Destructuring**" section completely?

Reply:
Yeah, I would say remove it completely, or say to use array destructuring when it is more readable

Reply:
I guess I was thinking a while back that maybe there should be some variation in the way some of these sections are structured, so that they aren't always a binary "Yep/Nope" setup. There could be examples of preferred style - in many cases - with caveats, as @calebeby discussed above. So I'm not sure the section needs to be scrapped, but maybe should have some more nuance. 

Also, @calebeby was specifically talking about ESLint rules, I think. Again, I think rules might be OK if they don't force someone into coding a particular way when it might not be the best way for certain scenarios.

Reply:
ESLint warnings will warn you about the problem but they won't cause the `eslint` CLI to exit with non-zero. I don't really like ESLint warnings because then you get squiggly lines in your editor for many things that aren't actually problems (and the only way to mark them as "addressed" is with `// eslint-disable-line ...`). My personal opinion is that we should not use ESLint warnings. I think in we should not have the `prefer-destructuring` rule enabled for arrays.

Reply:
Agreed. If there were a more polite setting for ESLint, I'd consider, but I also don't like having to create ESLint exceptions in my code for things I really want to be able to do.

As for the section, @gerardo-rodriguez, I'm still open to a more nuanced organizational style where you discuss conditional recommendations, if that makes sense.

Reply:
Thanks for the feedback @calebeby and @Josh68! 

I restructured this section to remove the Nope/Yep + updated the wording. I also removed the `prefer-destructuring` ESLint link for the array section. I agree that we should not use ESLint warnings and that we should not have the `prefer-destructuring` rule enabled for arrays. 

Let me know what you think. If the **4.2 Array Destructuring** section is _still_ confusing or not helpful based on the latest PR changes, I'll go ahead and remove it instead. I like your recommendation of not enforcing one over the other. 

Thanks again! 😉

Reply:
Rubber stamping!